dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -324,6 +333,92 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
+                String[] splitDocker = image.split("/");
+                String[] splitTag = splitDocker[2].split(":");
+                if (splitTag.length > 1) {
+                    String repo = splitDocker[1] + "/" + splitTag[0];
+                    String tagName = splitTag[1];
+                    response = getImageResponseFromQuay(image);
+
+                    if (response.isPresent()) {
+                        Map<String, ArrayList<Map<String, String>>> map = new HashMap<>();
+                        Map<String, String> errorMap = new HashMap<>();
+                        map = (Map<String, ArrayList<Map<String, String>>>)gson.fromJson(response.get(), map.getClass());
+                        errorMap = (Map<String, String>)gson.fromJson(response.get(), errorMap.getClass());
+                        if (errorMap.get("error_message") != null) {
+                            LOG.error("Error response from Quay: " + errorMap.get("error_message"));
+                        }
+
+                        try {
+                            final List<Map<String, String>> array = map.get("tags");
+
+                            for (Map<String, String> tag : array) {
+                                final String digest = tag.get("manifest_digest");
+                                final String imageID = tag.get("image_id");
+                                List<Checksum> checksums = new ArrayList<>();
+                                checksums.add(new Checksum(digest.split(":")[0], digest.split(":")[1]));
+                                dockerImages.add(new Image(checksums, repo, tagName, imageID));
+                            }
+
+                        } catch (IndexOutOfBoundsException | NullPointerException ex) {
+                            LOG.error("Could not get checksum information for " + splitDocker[1]);
+                        }
+                    }
+                } else {
+                    LOG.error("Could not find image version specified for " + splitDocker[1]);
+                }
+
+            } else if (image.startsWith("images.sbgenomics")) {
+                return dockerImages;
+
+            } else if (image.startsWith("registry.gitlab.com")) {
+                return dockerImages;
+
+            } else if (parts.length == 2) {
+                //dockerhub
+                return dockerImages;
+            }
+        }
+        return dockerImages;
+    }
+
+    default Optional<String> getImageResponseFromQuay(String dockerName) {
+        String[] splitDocker = dockerName.split("/");
+        String[] splitTag = splitDocker[2].split(":");
+        final String tagUrl = QUAY_URL + "repository/" + splitDocker[1] + "/" + splitTag[0] + "/tag/" + "?specificTag=" + splitTag[1];
+        Optional<String> response;
+
+        try {
+            URL url = new URL(tagUrl);
+            response = Optional.of(IOUtils.toString(url, StandardCharsets.UTF_8));
+            if (response.isPresent()) {
```

Comment:
I don't think you need this if statement; if it's empty, you would just return the empty optional, which is what you want.